operator/pkg/module/parse/merger.go

```diff
@@ -0,0 +1,40 @@
+package parse
+
+import (
+	"fmt"
+
+	"github.com/imdario/mergo"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/kyma-project/lifecycle-manager/operator/api/v1alpha1"
+)
+
+func mergeResourceIntoSpec(resource *unstructured.Unstructured, component *unstructured.Unstructured) error {
+	if err := mergo.Merge(&component.Object,
+		map[string]any{"spec": map[string]any{"resource": resource}},
+		mergo.WithAppendSlice); err != nil {
+		return fmt.Errorf("error while merging the template spec.data into the spec: %w", err)
+	}
+	return nil
+}
+
+// mergeTargetIntoSpec uses a given v1alpha1.Target to set the remote attribute on an unstructured.
+func mergeTargetIntoSpec(remote v1alpha1.Target, component *unstructured.Unstructured) error {
+	if err := mergo.Merge(&component.Object,
+		map[string]any{"remote": ConvertTargetToRemote(remote)},
+		mergo.WithAppendSlice); err != nil {
+		return fmt.Errorf("error while merging the remote attribute into the spec: %w", err)
+	}
+	return nil
+}
+
+func ConvertTargetToRemote(remote v1alpha1.Target) bool {
```

Comment:
I actually would just propagate this target to manifest, and introduce a new target value, like `TargetLocal`)  but since the logic in manifest already [implemented](https://github.com/kyma-project/module-manager/pull/89/files), I'm fine with it.
But do you think if it make sense to add a new target value, like `TargetLocal`? I think it make the kyma resource a bit intuitive to understand

Reply:
I think if we want to do that, this should be a separate PR for refactoring to not make it too big here. (it also changes the CRD after all and all samples)

Reply:
ok, created a proposal issue for tracking this, we can consider it later. https://github.com/kyma-project/lifecycle-manager/issues/167